core/stateful_precompile_test.go

```diff
@@ -1060,22 +941,321 @@ func TestFeeConfigManagerRun(t *testing.T) {
 			if len(test.expectedErr) != 0 {
-				if err == nil {
-					assert.Failf(t, "run expectedly passed without error", "expected error %q", test.expectedErr)
-				} else {
-					assert.True(t, strings.Contains(err.Error(), test.expectedErr), "expected error (%s) to contain substring (%s)", err, test.expectedErr)
-				}
-				return
+				require.ErrorContains(t, err, test.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+
+			require.Equal(t, uint64(0), remainingGas)
+			require.Equal(t, test.expectedRes, ret)
+
+			if test.assertState != nil {
+				test.assertState(t, state)
 			}
+		})
+	}
+}
+
+func TestRewardManagerRun(t *testing.T) {
+	type test struct {
+		caller       common.Address
+		preCondition func(t *testing.T, state *state.StateDB)
+		input        func() []byte
+		suppliedGas  uint64
+		readOnly     bool
+		config       *precompile.RewardManagerConfig
+
+		expectedRes []byte
+		expectedErr string
 
-			if err != nil {
-				t.Fatal(err)
+		assertState func(t *testing.T, state *state.StateDB)
+	}
+
+	adminAddr := common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC")
+	enabledAddr := common.HexToAddress("0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B")
+	noRoleAddr := common.HexToAddress("0xF60C45c607D0f41687c94C314d300f483661E13a")
+	testAddr := common.HexToAddress("0x0123")
+
+	for name, test := range map[string]test{
+		"set allow fee recipients from no role fails": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				input, err := precompile.PackAllowFeeRecipients()
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: precompile.AllowFeeRecipientsGasCost,
+			readOnly:    false,
+			expectedErr: precompile.ErrCannotAllowFeeRecipients.Error(),
+		},
+		"set reward address from no role fails": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				input, err := precompile.PackSetRewardAddress(testAddr)
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: precompile.SetRewardAddressGasCost,
+			readOnly:    false,
+			expectedErr: precompile.ErrCannotSetRewardAddress.Error(),
+		},
+		"disable rewards from no role fails": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				input, err := precompile.PackDisableRewards()
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: precompile.DisableRewardsGasCost,
+			readOnly:    false,
+			expectedErr: precompile.ErrCannotDisableRewards.Error(),
+		},
+		"set allow fee recipients from enabled succeeds": {
+			caller: enabledAddr,
+			input: func() []byte {
+				input, err := precompile.PackAllowFeeRecipients()
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: precompile.AllowFeeRecipientsGasCost,
+			readOnly:    false,
+			expectedRes: []byte{},
+		},
+		"set reward address from enabled succeeds": {
```

Comment:
Should we add an assertion on the state to the end of this test?

Reply:
added